### PR TITLE
Console improvements

### PR DESCRIPTION
**Pull request by etiennebarrie**
Just wanted to try out writing a test class in the console and run the tests, also found that having a method called `pretty_print` on `Megatest::Config` wasn't super nice for debugging in a console.

```console
$ bin/console                                                                                                                                                                                                                            
?> class FooTest < Megatest::Test
?>   test "fails" do
?>     assert false
?>   end
>> end
=> #<Megatest::State::BlockTest: FooTest#fails @ (irb):2>
>> test!
Running 1 test cases with --seed 60698

F

  1) Failure: FooTest#fails

  Expected false to be truthy

  (irb):3:in `block in <class:FooTest>'

megatest (irb):2

Ran 1 cases, 1 assertions, 1 failures, 0 errors, 0 retries, 0 skips
=> false
```

I believe the `test!` method for the console (modeled after `reload!` in Rails) shows that we could refactor the classes a bit but I'm not sure which way exactly yet:
* we could have a registry by default to avoid having to wrap any code defining tests with `Megatest.with_registry`
* `build_queue` could be on `Megatest` directly, and maybe the logic around distributed sharded could be done there
* building the reporters could be done by the config and moved out of CLI (`verbose` and `junit` could be part of the config too, as well as `@out` and `@err`)
* `CLI#executor` could be moved to `Megatest`

### CODE REVIEW

**Comment by byroot** [comment]
> * having to wrap any code defining tests with `Megatest.with_registry`

Yeah, we'll probably have to do something like that to support `autorun` mode (e.g. for bundler/inline repro scripts).

I did it this way at first to be strict. But we could have a registry by default, and then replace it by something that yells at you if you define any more tests once the runner has started running the suite.

File: lib/megatest/config.rb [hunk 1]
```diff
@@ -214,10 +214,9 @@ def diff(expected, actual)
       @differ&.call(expected, actual)
     end
 
-    def pretty_print(object)
+    def print_pretty(object)
```
Comment (byroot) [inline lib/megatest/config.rb:217]: A bit rougth of a name, no? Maybe `render_object` or something?

**Comment by etiennebarrie** [comment]
> something that yells at you if you define any more tests once the runner has started running the suite

Right but that would make it annoying for the console, if you want to add more tests after doing a run. Not a big deal though, it's only for us contributors to test small things, we can always restart the console.

**Comment by casperisfine** [comment]
> that would make it annoying for the console, if you want to add more tests after doing a run

We can special case it in the console. I just want to catch the case in real world scenario where some test files are somehow lazy loaded and end up being ignored.

**Pull request merged**